src/components/Editor/components/ViewportAwareButton.js

```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import { Button } from "reactstrap";
+
+/**-------Props--------
+ * isSmall: dictates display of text field
+ * icon: desired icon (i.e. <FontAwesomeIcon icon={...} />)
+ * text: the text label for the button
+ */
+/**--------Optional props--------
+ * any props you would apply to a reactstrap Button.
+ */
+const ViewportAwareButton = props => {
+  // extract icon and text from props.
+  let { icon, text, ...remainder } = props;
+
+  // render icon always, text only if not small.
+  return (
+    <Button {...remainder}>
+      {icon}
```

Comment:
Sorry to be super nit-picky here, but I don't think you should pass down the entire `<FontAwesomeIcon>` as the prop for `icon`; the `ViewportAwareButton` should be the one that handles importing `<FontAwesomeIcon>` and inserting the resulting icon in this file. I think this makes the code a little clearer and slightly easier to use, at maybe the expense of future component flexibility.

@fan-matt any thoughts on this?

Reply:
Here's my 2 cents on the issue:

I think you shouldn't pass in any children in as props since this makes the button very limited in what it looks like. To maximize our "drop-in replacement" theme we're going for, I think it makes sense to do something like this:

```javascript
// App.js
import React from 'react';

import Container from './Container';


function App() {
  return (
    <div className="App">
      <Container>
        <h1 className="a">I'm A</h1>
        <h1 className="b">I'm B</h1>
      </Container>
    </div>
  );
}

export default App;


// Container.js
import React from "react";

export default class Container extends React.Component {

    render() {
        console.log(this.props.children);

        let renderMe = this.props.children.filter( (child) => {
            return child.props.className === "a";
        });

        return(
            <div>
                {renderMe}
            </div>
        );
    }
}
```
This only renders "I'm A"

Reply:
> it definitely sits better with me to write it like the previous iteration

+1 on this. I am flexible in how to implement this, but I thought the first iteration made a little more sense since one could use it as a "drop-in" replacement. It isn't necessarily a requirement that every `<ViewportAwareButton>` have an icon, just like how it isn't a requirement that every `<Button>` have one. `<Button>`s have bodies and children, as should their ViewportAware counterparts. However, it is a possibility that one would still desire to make certain components within said Button vanish when the screen gets too small, in which case a ViewportAware flavor is preferable.

> passing down the entire `<FontAwesomeIcon>` since it future-proofs this component

That was the logic in my decision there. We are using FontAwesome for the time being, but this is something that could change later down the line - hence passing the entire component as a prop. While I too am not a fan of passing the `icon` prop down this way, I figured it would be best for future-proofing as Matt mentioned.